drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -424,12 +430,83 @@ int lis2dh_init(const struct device *dev)
 		    cfg->bus_name, 1 << (LIS2DH_FS_IDX + 1),
 		    LIS2DH_ODR_IDX, (uint8_t)LIS2DH_LP_EN_BIT, lis2dh->scale);
 
+#ifdef CONFIG_PM_DEVICE
+	lis2dh->power_state = PM_DEVICE_STATE_ACTIVE;
+	lis2dh->reg_ctrl1_active_val = LIS2DH_ACCEL_EN_BITS | LIS2DH_LP_EN_BIT |
+				       LIS2DH_ODR_BITS;
+#endif
+
 	/* enable accel measurements and set power mode and data rate */
 	return lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL1,
 					LIS2DH_ACCEL_EN_BITS | LIS2DH_LP_EN_BIT |
 					LIS2DH_ODR_BITS);
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int lis2dh_set_power_state(const struct device *dev,
+				  enum pm_device_state new_state)
+{
+	struct lis2dh_data *lis2dh = dev->data;
+	int status = 0;
+
+	if (new_state == PM_DEVICE_STATE_ACTIVE) {
+		status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL1,
+						  lis2dh->reg_ctrl1_active_val);
+		if (status) {
+			LOG_ERR("setting active state failed");
+		}
+		lis2dh->power_state = PM_DEVICE_STATE_ACTIVE;
+		LOG_INF("state changed to active");
+	} else {
+		__ASSERT_NO_MSG(new_state == PM_DEVICE_STATE_LOW_POWER ||
+				new_state == PM_DEVICE_STATE_SUSPEND ||
+				new_state == PM_DEVICE_STATE_OFF);
```

Comment:
Please don't use this pattern. Each state requires different actions. In general, you just need to add support for `SUSPEND`. `LOW_POWER` should actually be removed (I thought it was?) and OFF means you're effectively cutting power.

Reply:
I've removed this and reconsidered this whole PR based on the current implementation.